src/server.c

```diff
@@ -3922,6 +3922,23 @@ int processCommand(client *c) {
         return C_OK;
     }
 
+    /* If the server is paused, block the client until
+     * the pause has ended. Replicas are never paused. */
+    int is_readonly_command = (c->cmd->flags & CMD_READONLY) ||
+                              (c->cmd->proc == execCommand && !(c->mstate.cmd_inv_flags & CMD_READONLY));
+
+    if (!(c->flags & CLIENT_SLAVE) && 
+        ((server.client_pause_flags & CLIENT_PAUSE_ALL) ||
+        (server.client_pause_flags & CLIENT_PAUSE_RO && 
+            !(is_readonly_command || 
```

Comment:
I don't think we wanna check for `CMD_READONLY`, i think we should check for `!CMD_WRITE`.
Seems to me that we do wanna allow non-keyspace commands like PING, INFO, CLIENT, SHUTDOWN, CONFIG etc.
And even basically any command that's not writing to the keyspace, like DBSIZE.
other examples off the to of my head (not already mentioned above): ACL, DEBUG, RESET, WATCH, DISCARD, SELECT, SYNC, SUBSCRIBE and MULTI / EXEC (both already handled in your code explicitly).

I suppose PUBLISH is ok too (doesn't propagate to replicas, so it's ok, right?)

The tricky part is EVAL (which we don't know it's content, and PFCOUNT (which is a ready-only, which **does** increment server.dirty).
i suppose we can explicitly block EVAL for now, and either add some code to PFCOUNT that will avoid the conversion and propagation, or block it too.
Modules already have the `RM_AvoidReplicaTraffic` (which i think you should fix to only look at the RO pause flag)

Reply:
Yeah, I was hoping this might save some effort, but while talking it through internally !Write is probably the way to go.

Reply:
Looks like this further relates to the PR you just put up as well.